src/frontend/hooks/blog-form.js

```diff
@@ -0,0 +1,201 @@
+import api from '../api';
+
+const useBlogForm = () => {
+  const [blogTitle, setblogTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [summary, setSummary] = useState('');
+  const [date, setDate] = useState('');
+  const [link, setLink] = useState('');
+  const [currentImageURL, setCurrentImageURL] = useState('');
+  const [uploadedImageURL, setUploadedImageURL] = useState('');
+  const [closed, setClosed] = useState(false);
+  const [visibility, setVisibility] = useState('Public');
+  const [category, setCategory] = useState('');
+  const [imagesToDelete, setImagesToDelete] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [errors, setErrors] = useState([]);
+  const history = useHistory();
+  const params = useParams();
+  const { blogInfo } = useBlogInfo();
+
+  useEffect(() => {
+    if (blogInfo.posts?.length > 0) {
+      (async () => {
+        try {
+          if (params.blogId) {
+            const blogId = parseInt(params.blogId, 10);
+            const blogToEdit = blogInfo.posts.find((blog) => blog.id === blogId); 
+            if (blogToEdit) {
+              setblogTitle(blogToEdit.title); 
+              setAuthor(blogToEdit.author);
+              setSummary(blogToEdit.summary);
+              setDate(blogToEdit.date);
+              setLink(blogToEdit.link);
+              setCurrentImageURL(blogToEdit.image); // direct link to google storage
+              setClosed(blogToEdit.closed);
+              setVisibility(blogToEdit.visibility);
+              setCategory(blogToEdit.category);
+            }
+          }
+        } catch (err) {
+          console.log(err);
+        }
+      })();
+    }
+  }, [setblogTitle, setAuthor, setSummary, setDate, setLink, 
+      setCurrentImageURL, setClosed, setVisibility, setCategory]);
+
+  const imageUpload = useCallback(
+    (imageUpload) => {
+      setUploadedImageURL(imageUpload);
+      setCurrentImageURL('');
+    },
+    [setUploadedImageURL],
+  );
+
+  const imageDelete = () => {
+    setUploadedImageURL('');
+    setCurrentImageURL('');
+  } 
+
+  const openModal = () => {
+    setShowModal(true);
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+
+  const closeForm = useCallback(() => {
+    history.push('/blog-posts');
+  }, [history]);
+
+  const allFieldsAreFilled = () => {
+    let tmpErrors = []
+
+    // define errors that will be sent to the user
+    if (blogTitle === "") { tmpErrors.push("title cannot be blank") }
+    if (author === "") { tmpErrors.push("author cannot be blank") }
+    if (summary === "") { tmpErrors.push("summary cannot be blank") }
+    if (link === "") { tmpErrors.push("link cannot be blank") }
+    if (date === "") { tmpErrors.push("date cannot be blank") }
+    if (category === "") {  tmpErrors.push("category cannot be blank") }
+
+    if (currentImageURL == '' && uploadedImageURL == '') {
+          tmpErrors.push( "One image is required")
```

Comment:
`currentImageURL` and `uploadedImageURL` are both initialized to be `''`. If a user is editing an existing blog (which will already have an image) then in the UseEffect it will be initialized to the image link. If the user is creating a new blog then `currentImageURL` will remain `''` since there's no image yet. 

Once the user uploads a new blog image, their current image (if it exists) will be removed (set to `''`) and the new image link will be stored. When they submit, either one of `currentImageURL` or `uploadedImageURL` will hold the image, which is valid, or both will be empty, which is invalid. As long as one of them has an image, they're good to go. That's why its an `and` condition, since both need to be empty to cause an error. Let me know if there's a flaw in this logic that I'm not seeing